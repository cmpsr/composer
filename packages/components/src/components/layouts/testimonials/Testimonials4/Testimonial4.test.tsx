import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Testimonials4, TESTIMONIAL4_PLAN_DEFAULT_TEST_ID } from '.';
import {
  IMAGE_DEFAULT_TEST_ID,
  ICON_DEFAULT_TEST_ID,
} from 'components/primitives';

describe('Testimonials4', () => {
  const asset = {
    contentType: 'image',
    title: 'Image Title',
    url:
      'https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg',
  };
  const testimonial =
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae aut, omnis, maxime rem, cum similique repellendus hic impedit totam magnam deserunt voluptatum quia! Blanditiis repellendus sit debitis doloribus, totam minima?';
  const author = 'Testing author,';
  const profession = 'Testing profession.';

  it('should match snapshot', () => {
    const testimonial4 = renderer.create(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const tree = testimonial4.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial4 = getByTestId(TESTIMONIAL4_PLAN_DEFAULT_TEST_ID);
    expect(testimonial4).toBeInTheDocument();
  });
  it('should render className', () => {
    const { getByTestId } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
        className="foo"
      />,
    );

    const testimonial4 = getByTestId(TESTIMONIAL4_PLAN_DEFAULT_TEST_ID);
    expect(testimonial4.classList).toContain('foo');
  });
  it('should render rounded asset', () => {
    const { getByTestId } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const assetComponent = getByTestId(IMAGE_DEFAULT_TEST_ID);
    expect(assetComponent).toBeInTheDocument();
    expect(assetComponent.classList).toContain('rounded-full');
  });
  it('should render testimonial', () => {
    const { getByText } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial4 = getByText(testimonial);
    expect(testimonial4).toBeInTheDocument();
  });
  it('should render author', () => {
    const { getByText } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial4 = getByText(author);
    expect(testimonial4).toBeInTheDocument();
  });
  it('should render profession', () => {
    const { getByText } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial4 = getByText(profession);
    expect(testimonial4).toBeInTheDocument();
  });
  it('should render svg icon', () => {
    const { getByTestId } = render(
      <Testimonials4
        asset={asset}
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const icon = getByTestId(ICON_DEFAULT_TEST_ID);
    expect(icon).toBeInTheDocument();
  });
});
