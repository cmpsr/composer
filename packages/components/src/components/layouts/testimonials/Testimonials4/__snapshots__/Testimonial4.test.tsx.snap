// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials4 should match snapshot 1`] = `
<div
  className="flex flex-col p-10 rounded shadow-lg"
  data-testid="testimonial4"
>
  <div
    className="flex items-center mb-8"
  >
    <picture
      className="select-none mr-6"
    >
      <source
        className="select-none w-20 h-20 rounded-full object-cover"
        data-testid="image"
        srcSet="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg?fm=webp"
        type="image/webp"
      />
      <img
        alt="Image Title"
        className="select-none w-20 h-20 rounded-full object-cover"
        src="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg"
      />
    </picture>
    <div
      className="align-middle"
    >
      <h6
        className="typo-headline-6"
        data-testid="typography"
      >
        Testing author,
      </h6>
      <p
        className="typo-body text-gray-500"
        data-testid="typography"
      >
        Testing profession.
      </p>
    </div>
  </div>
  <p
    className="typo-body text-left mb-12"
    data-testid="typography"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae aut, omnis, maxime rem, cum similique repellendus hic impedit totam magnam deserunt voluptatum quia! Blanditiis repellendus sit debitis doloribus, totam minima?
  </p>
  <div
    className="flex justify-end"
  >
    <svg
      data-testid="icon"
      height="50px"
      viewBox="0 0 24 24"
      width="50px"
    >
      <path
        d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
      />
    </svg>
  </div>
</div>
`;
